tests/acceptance/01_vars/02_functions/string.cf

```diff
@@ -0,0 +1,56 @@
+body common control
+{
+	bundlesequence => { "init", "check" };
+}
+
+bundle agent init
+{
+	classes:
+		"ExistingClass";
+
+	vars:
+		"aInt" int => "1k";
+		"aReal" real => "6.066";
+		"aString" string => "Really interesting string";
+		"aIList" ilist => {1, 2, 3};
+		"aSList" slist => {
+			"ExistingClass",
+			string(and("ExistingClass", strcmp("bca", "abc")))
+		};
+
+		"convInt" string => string("$(aInt)");
+		"convReal" string => string("$(aReal)");
+		"convString" string => string("$(aString)");
+		"convBoolTrue" string => string(and("ExistingClass"));
+		"convBoolFalse" string => string(and("NonExistingClass"));
+		"convRandInt" string => string(randomint(10, 1000));
+
+#		"testArgs" string => string();  # This is caught by the parser
+}
+
+bundle agent check
+{
+	vars:
+		"thisfilename" string => "$(this.promise_filename)";
+	classes:
+		"okInt" expression => strcmp("$(init.aInt)", "$(init.convInt)");
+		"okReal" expression => strcmp("$(init.aReal)", "$(init.convReal)");
+		"okString" expression => strcmp("$(init.aString)", "$(init.convString)");
+		"okBoolTrue" expression => strcmp("any", "$(init.convBoolTrue)");
+		"okBoolFalse" expression => strcmp("!any", "$(init.convBoolFalse)");
+		"okALL" expression => and("okInt", "okReal", "okString", "okBoolTrue", "okBoolFalse");
+
+	reports:
+		!okInt::
+			"$(thisfilename) int FAIL";
+		!okReal::
+			"$(thisfilename) real FAIL";
+		!okString::
+			"$(thisfilename) string FAIL";
+		!okBoolTrue::
+			'$(thisfilename) bool "any" FAIL';
+		!okBoolFalse::
+			'$(thisfilename) bool "!any" FAIL';
+		okALL::
+			"$(thisfilename) pass";
```

Comment:
Change this to 2 messages, one for `ok` and one for `!ok`:

https://github.com/cfengine/core/blob/master/tests/acceptance/README#L159-L160

https://github.com/cfengine/core/blob/5d361487ce2da4d61c6fad81afbe22806ca9f5b2/tests/acceptance/30_custom_promise_types/11_unless.cf#L125-L128